tests/test_resume.rs

```diff
@@ -242,10 +242,22 @@ impl MachineTy {
     }
 }
 
+struct ConstantCyclesCtx;
+
+impl<Mac: SupportMachine> ExecutionContext<Mac> for ConstantCyclesCtx {
```

Comment:
Similar to the above

Reply:
This is when you need to name the context type. This is personally preferable to `WithCyclesFunc<(), fn ...>` or `Box dyn`.

Reply:
I don't really get what you want the context type to look like here. `impl ExecutionContext for F where F: Fn(Instruction) -> u64` and then `SomeMachine<fn(Instruction) -> u64>`? Doesn't look better than this IMO.